components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -157,6 +157,17 @@ public BuildValidationData fetchBuildValidationData(String buildScanId) {
                     throw new BuildScanNotFoundException(buildScanId, baseUrl, e.getCode(), e.getResponseBody(), e);
                 case StatusCodes.UNAUTHORIZED:
                     throw new AuthenticationFailedException(buildScanId, baseUrl, e.getCode(), e.getResponseBody(), e);
+                case 0:
+                    // This wasn't a bad HTTP response, instead it was a failure to connect.
+                    //
+                    // WORKAROUND: There is a bug in the OpenAPI generator that generates a broken ApiException.getMessage(),
```

Comment:
❔ Is there an open issue for OpenAPI generator for this that we can link here?

Reply:
It seems it's https://github.com/OpenAPITools/openapi-generator/issues/12631 and there should already be a fix for that on latest `master`.

Reply:
Added a link to the issue to the comment.